### PR TITLE
Put test split files in test-splits dir

### PR DESCRIPTION
**Pull request by blindpirate**
So that they're not cleaned up.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
Saw the changes in https://github.com/gradle/gradle/pull/14511/files

**Comment by wolfs** [comment]
@blindpirate The changes in https://github.com/gradle/gradle/pull/14511 are somewhat orthogonal to the changes here, since they use a different file for performance test splits. I think it would be good to add your PR first.

**Comment by blindpirate** [comment]
Yes, I misread.

File: buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/integrationtests/BuildBucketProvider.kt [hunk 1]
```diff
@@ -31,12 +31,12 @@ fun Project.bucketProvider(): BuildBucketProvider {
     if (!rootProject.extra.has("bucketProvider")) {
         rootProject.extra["bucketProvider"] = when {
             project.stringPropertyOrEmpty("includeTestClasses").isNotBlank() -> {
-                val content = project.rootProject.buildDir.resolve("include-test-classes.properties").readText()
+                val content = project.rootProject.rootDir.resolve("test-splits/include-test-classes.properties").readText()
```
Comment (wolfs) [inline buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/integrationtests/BuildBucketProvider.kt:34]:
```suggestion
                val content = project.rootProject.file("test-splits/include-test-classes.properties").readText()
```

File: .teamcity/Gradle_Check/model/BucketProvider.kt [hunk 2]
```diff
@@ -239,28 +239,28 @@ class LargeSubprojectSplitBucket(val subproject: GradleSubproject, val number: I
         val testClasses = classes.map { it.toPropertiesLine() }
         val action = if (include) "include" else "exclude"
         val unixScript = """
-mkdir -p build
-rm -rf build/*-test-classes.properties
-cat > build/$action-test-classes.properties << EOL
+mkdir -p test-splits
+rm -rf test-splits/*-test-classes.properties
```
Comment (wolfs) [inline .teamcity/Gradle_Check/model/BucketProvider.kt:243]: Should we delete the whole `test-splits` directory before adding the new `-test-classes.properties`?

**Review by wolfs — CHANGES_REQUESTED** [review]
Looks good! I added some comments.

File: .teamcity/Gradle_Check/model/BucketProvider.kt (hunk 2 above)
Comment (blindpirate) [reply .teamcity/Gradle_Check/model/BucketProvider.kt:243]: No need I think...

File: buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/integrationtests/BuildBucketProvider.kt [hunk 3]
```diff
@@ -31,12 +31,12 @@ fun Project.bucketProvider(): BuildBucketProvider {
...
         rootProject.extra["bucketProvider"] = when {
             project.stringPropertyOrEmpty("includeTestClasses").isNotBlank() -> {
-                val content = project.rootProject.buildDir.resolve("include-test-classes.properties").readText()
+                val content = project.rootProject.file("test-splits/include-test-classes.properties").readText()
                 println("Tests to be included:\n$content")
                 IncludeTestClassProvider(readTestClasses(content))
             }
             project.stringPropertyOrEmpty("excludeTestClasses").isNotBlank() -> {
-                val content = project.rootProject.buildDir.resolve("exclude-test-classes.properties").readText()
+                val content = project.rootProject.rootDir.resolve("test-splits/exclude-test-classes.properties").readText()
```
Comment (blindpirate) [inline buildSrc/subprojects/integration-testing/src/main/kotlin/gradlebuild/integrationtests/BuildBucketProvider.kt:39]:
```suggestion
                val content = project.rootProject.file("test-splits/exclude-test-classes.properties").readText()
```

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Pull request merged**